Patch-Source: <https://gitlab.freedesktop.org/pipewire/pipewire/-/merge_requests/2717>
---
From d98e46636ef615801bcb25dee3ec1aa4a8e7344d Mon Sep 17 00:00:00 2001
From: Achill Gilgenast <achill@achill.org>
Date: Tue, 3 Feb 2026 22:49:16 +0100
Subject: [PATCH] systemd: Allow installation of systemd services without
 libsystemd

Allows installation of systemd services without libsystemd installed.
Useful for Alpine Linux where systemd services are allowed to be subpackaged
(e.g. for postmarketOS) but hasn't systemd in it's repos.

It has no change in existing behavior, but installs services if the unit
directories are explicitly set.

Signed-off-by: Achill Gilgenast <achill@achill.org>
---
 src/daemon/systemd/system/meson.build | 13 +++++++++----
 src/daemon/systemd/user/meson.build   | 13 +++++++++----
 2 files changed, 18 insertions(+), 8 deletions(-)

diff --git a/src/daemon/systemd/system/meson.build b/src/daemon/systemd/system/meson.build
index 02efc7a41042..329e165318c4 100644
--- a/src/daemon/systemd/system/meson.build
+++ b/src/daemon/systemd/system/meson.build
@@ -1,13 +1,18 @@
-systemd = dependency('systemd', required : get_option('systemd-system-service'))
-if not systemd.found()
-  subdir_done()
+systemd_system_services_dir = ''
+
+systemd = dependency('systemd', required : false)
+if systemd.found()
+  systemd_system_services_dir = systemd.get_variable('systemdsystemunitdir', pkgconfig_define : [ 'rootprefix', prefix])
 endif
 
-systemd_system_services_dir = systemd.get_variable('systemdsystemunitdir', pkgconfig_define : [ 'rootprefix', prefix])
 if get_option('systemd-system-unit-dir') != ''
   systemd_system_services_dir = get_option('systemd-system-unit-dir')
 endif
 
+if systemd_system_services_dir == ''
+  subdir_done()
+endif
+
 install_data(sources : ['pipewire.socket', 'pipewire-manager.socket', 'pipewire-pulse.socket' ],
              install_dir : systemd_system_services_dir)
 
diff --git a/src/daemon/systemd/user/meson.build b/src/daemon/systemd/user/meson.build
index 1b65d5f7aacf..fcf8c39129da 100644
--- a/src/daemon/systemd/user/meson.build
+++ b/src/daemon/systemd/user/meson.build
@@ -1,13 +1,18 @@
-systemd = dependency('systemd', required : get_option('systemd-user-service'))
-if not systemd.found()
-  subdir_done()
+systemd_user_services_dir = ''
+
+systemd = dependency('systemd', required : false)
+if systemd.found()
+  systemd_user_services_dir = systemd.get_variable('systemduserunitdir', pkgconfig_define : [ 'prefix', prefix])
 endif
 
-systemd_user_services_dir = systemd.get_variable('systemduserunitdir', pkgconfig_define : [ 'prefix', prefix])
 if get_option('systemd-user-unit-dir') != ''
   systemd_user_services_dir = get_option('systemd-user-unit-dir')
 endif
 
+if systemd_user_services_dir == ''
+  subdir_done()
+endif
+
 install_data(
   sources : ['pipewire.socket', 'pipewire-pulse.socket'],
   install_dir : systemd_user_services_dir)
-- 
2.53.0

